GVFS/GVFS.Common/Maintenance/GitMaintenanceStep.cs

```diff
@@ -143,6 +149,47 @@ protected EventMetadata CreateEventMetadata(Exception e = null)
             return metadata;
         }
 
+        protected bool EnoughTimeBetweenRuns()
+        {
+            if (!this.Context.FileSystem.FileExists(this.LastRunTimeFilePath))
+            {
+                return true;
+            }
+
+            string lastRunTime = this.Context.FileSystem.ReadAllText(this.LastRunTimeFilePath);
+            if (!DateTime.TryParseExact(

```

Comment:
This would be so much simpler if you used `EpochConverter` to store a unix-style timestamp.

Reply:
I updated it to use EpochConverter is a seperate commit.

I honestly still prefer the formatter way, but this will work to.  Just add the unit tests as promised ;-)

Reply:
#620

Reply:
I should add context here as to why I didn't love using the EpochConverter
- I like having readable datetimes in files.  This easily lets me verify when the last run was.  By saving a long we have to use a converter.
- EpochConverter has two functions ToUnixEpochSeconds/FromUnixEpochSeconds, however only one of them is in use (FromUnixEpochSeconds in Prefetch).  @derrickstolee has since adding unit tests so we know both functions work.
- Since we're redoing the way we look up last Prefetch in the future, I saw that as opportunity to nix the EpochConverter and switch everything to readable formated dates
- In high performance situations saving a long rather than a datetime makes a ton of sense.  It'll be faster for comparisons (+/- of long is faster than dates).  However, we're not doing that here.
- Prefetch files use long for datetime so using that method has consistency with the CacheServer
- The file timestamp should also be a clue to the last runtime, however we'd still need to convert a long to be sure